### PR TITLE
feat(Overlay): Adds Overlay body scroll lock [GM-26]

### PR DESCRIPTION
**Pull request by codecaaron**
## Overview
Currently you are able to scroll through background content while an overlay is open leading to some wonky UX.  This change locks the body scrolling to prevent a strange experience while an overlay is open.

### PR Checklist

- [ ] Related to Abstract designs:
- [x] Related to JIRA ticket: GM-26
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### Description
- Adds `useLockBodyScroll` to Overlay that is toggled on `isOpen`
- adds `react-use` as a blanket dependency for `gamut`
---

## Merging your changes

When you are ready to merge to master and publish your changes, use the "squash and merge" button in GitHub, and follow the [PR Title Guide](https://github.com/Codecademy/client-modules#pr-title-guide) to format your PR title and write your PR merge description.

**IMPORTANT:** If your PR contains breaking changes, please remember to follow the instructions for breaking changes!

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.35-alpha.c8d410.0<br/>@codecademy/gamut@9.2.8-alpha.c8d410.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5efd250a080b4601f01f2ccf--codecademy-styleguide.netlify.app>https://5efd250a080b4601f01f2ccf--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5efd250a080b4601f01f2ccf>Deploy Logs</a></p>

File: packages/gamut/src/Overlay/index.tsx [hunk 1]
```diff
@@ -34,8 +34,12 @@ export const Overlay: React.FC<OverlayProps> = ({
 }) => {
   if (!isOpen) return null;
 
+  const body = document.querySelector('body');
+  const onOpen = () => body.classList.add(styles.modalOpen);
+  const onClose = () => body.classList.remove(styles.modalOpen);
+
   return (
-    <BodyPortal>
+    <BodyPortal onMount={onOpen} onDismount={onClose}>
```
Comment (jakemhiller) [inline packages/gamut/src/Overlay/index.tsx:42]: we have `react-use` in the dependencies, so we could probably use this if we want to simplify things: https://github.com/streamich/react-use/blob/master/src/useLockBodyScroll.ts

it's a little... complex, compared to just using CSS/Class Names, but it seems like they've considered some accessibility/mobile edge cases.

Comment (codecaaron) [reply packages/gamut/src/Overlay/index.tsx:42]: IIRC we don't have it in client modules yet.  Do we want to include it?

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@3.1.0-alpha.dca022.0<br/>@codecademy/gamut-labs@1.8.8-alpha.dca022.0<br/>@codecademy/gamut@12.3.0-alpha.dca022.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f31d140edec8d013b172f70--gamut-preview.netlify.app>https://5f31d140edec8d013b172f70--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f31d140edec8d013b172f70>Deploy Logs</a></p>

File: packages/gamut/src/Overlay/index.tsx (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/gamut/src/Overlay/index.tsx]: My vote is: yes 😄

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@3.1.0-alpha.e89b18.0<br/>@codecademy/gamut-labs@1.8.8-alpha.e89b18.0<br/>@codecademy/gamut@12.3.0-alpha.e89b18.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f31d20c5852010093d21d44--gamut-preview.netlify.app>https://5f31d20c5852010093d21d44--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f31d20c5852010093d21d44>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f31d215edec8d013b172f99--gamut-preview.netlify.app>https://5f31d215edec8d013b172f99--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f31d215edec8d013b172f99>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@3.1.0-alpha.231de2.0<br/>@codecademy/gamut-labs@1.8.8-alpha.231de2.0<br/>@codecademy/gamut@12.3.0-alpha.231de2.0<br/></pre>

**Pull request merged**